core/domain/user_jobs_one_off.py

```diff
@@ -81,6 +81,41 @@ def reduce(key, version_and_exp_ids):
                     edited_exploration_ids))
 
 
+class UserAuthModelOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for creating and populating UserAuthModel for
+    all registered users.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(item):
+        """Implements the map function for this job."""
+        user_id = item.id
+        gae_id = [item.gae_id]
+        user_auth_model = user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, gae_id[0])
+
+        # For current migration, one user_id will strictly have one gae_id
+        # associated with it and vice versa.
+        if user_auth_model is None:
+            user_models.UserAuthModel(
+                id=user_id,
+                gae_id=gae_id
+            ).put()
+        elif user_auth_model.id != user_id:
+            # Because <user_id, gae_id> tuple coming from UserSettingsModel
```

Comment:
Doesn't this case mean that a GAE ID is being shared between multiple users? I think we should trigger a failure in this case since it should never happen.

Reply:
A GAE ID can be shared between one user an its profiles. Since they are stored as user models they can be shared, however in our migration we should verify that two full accounts do not share the same gae_id, only profiles and their corresponding full account can share it.

Reply:
The code and the tests for it have been modified now. PTAL!

Reply:
That association doesn't actually make sense to me. See my other comment thread--I think this is going to cause problems when we actually start using the auth model for login.